google-cloud-datastore/lib/google/cloud/datastore/service.rb

```diff
@@ -47,7 +49,7 @@ def service
             config.endpoint = host if host
             config.lib_name = "gccl"
             config.lib_version = Google::Cloud::Datastore::VERSION
-            config.metadata = { "google-cloud-resource-prefix": "projects/#{@project}" }
+            config.metadata = { "google-cloud-resource-prefix": "projects/#{@project}/databases/#{database}" }
```

Comment:
What happens if `database` is nil?

Reply:
Where does it default to the value `(default)`? As I'm looking at the code, it looks like it will default to the empty string. (In [google-cloud-datastore/lib/google/cloud/datastore.rb](https://github.com/googleapis/google-cloud-ruby/pull/20038/files#diff-1fbc549fc64984da75828c68f550d040c11d962df84dab257250432a0d412f2d) the value passed into the Service constructor defaults to `configure.database_id`, which defaults to the empty string in [google-cloud-datastore/lib/google-cloud-datastore.rb](https://github.com/googleapis/google-cloud-ruby/pull/20038/files#diff-48a53bb9345da22863041498e11647063dd8052a36cec9f96e13ace08b8921bd). So it seems like you'll get resource prefixes of the form `projects/my-project/databases/`. Is that correct?

Reply:
In fact, now that I look at your firestore PR ([google-cloud-firestore/lib/google-cloud-firestore.rb](https://github.com/googleapis/google-cloud-ruby/pull/20029/files#diff-ad05b08603d1aa178454bef8b5d8913cd4e136f48b683ebab463b937333c9d17)), the default configuration value there is `"(default)"` whereas here it's the empty string `""`.

Reply:
> In fact, now that I look at your firestore PR ([google-cloud-firestore/lib/google-cloud-firestore.rb](https://github.com/googleapis/google-cloud-ruby/pull/20029/files#diff-ad05b08603d1aa178454bef8b5d8913cd4e136f48b683ebab463b937333c9d17)), the default configuration value there is `"(default)"` whereas here it's the empty string `""`.

Yes in Firestore native mode we are allowed to use `(default)` as database name but not in Datastore. More details - https://github.com/googleapis/google-cloud-ruby/blob/main/google-cloud-datastore-v1/lib/google/cloud/datastore/v1/datastore/client.rb#L201